SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularitySlaveAndRackManager.java

```diff
@@ -199,13 +204,22 @@ public void slaveLost(SlaveID slaveIdObj) {
 
     if (slave.isPresent()) {
       slaveManager.changeState(slave.get(), MachineState.DEAD, Optional.<String> absent(), Optional.<String> absent());
+      if (configuration.getDisasterDetection().isEnabled()) {
+        updateDisasterCounter(slave.get());
+      }
 
       checkRackAfterSlaveLoss(slave.get());
     } else {
       LOG.warn("Lost a slave {}, but didn't know about it", slaveId);
     }
   }
 
+  private void updateDisasterCounter(SingularitySlave slave) {
+    if (slave.getCurrentState().getState() == MachineState.ACTIVE) {
```

Comment:
Ah, yeah the variable could probably use renaming. The one I'm grabbing is actually the previous slave state, not the one we are about to update to. Essentially I'm looking for any transitions from `ACTIVE` -> `DEAD`.